app/controllers/api/v1/deals_controller.rb

```diff
@@ -0,0 +1,8 @@
+module Api::V1
+  class DealsController < ApplicationController
+    def index
+      @deals = Deal.all
```

Comment:
I created a new route with pagination for when I am trying to get data for deals to show in a table.
Otherwise, I only hit this route when I get data for a quick search front end component. I've seen examples where the controller method takes in a query and returns data, but right now I return all deals, only once, and then use that data on the front end. Instead of hitting this multiple times whenever the user search changes. What do you think? How would you implement a quick search on the front end. Might be a good question for Jeff, too.

Reply:
im not sure what you mean by quick search, you mean like the one we have on the site?  That is elastic search, lots of ways to impliment but - i wouldn't recommend using postgres for that.  Lets not worry about quick search for now daniel.  lets stay focused on rails and react.